### ISSUE TITLE
DBT run fails and throws a TrinoExternalError in QA

### ISSUE THREAD

**Issue by shaidar** [mitodl/ol-data-platform#557]
### Steps to Reproduce

Running dbt on QA models, throws a TrinoExternalError.
Here is an example `dbt  --debug run --full-refresh --select models/staging/micromasters/stg__micromasters__app__postgres__auth_user.sql`

### Expected Behavior

(Optional)

### Actual Behavior

(Optional)

### Stacktrace

```
17:17:31.077413 [debug] [MainThread]: Sending event: {'category': 'dbt', 'action': 'deprecation', 'label': '23ee032b-8d56-4a19-b107-2fffc83e9042', 'property_': 'warn', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x10c27ddf0>]}
17:17:31.664230 [debug] [MainThread]: Partial parsing enabled: 0 files deleted, 0 files added, 0 files changed.
17:17:31.664532 [debug] [MainThread]: Partial parsing enabled, no changes found, skipping parsing
17:17:31.896153 [debug] [MainThread]: Sending event: {'category': 'dbt', 'action': 'load_project', 'label': '23ee032b-8d56-4a19-b107-2fffc83e9042', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x10f15d0d0>]}
17:17:32.668848 [debug] [MainThread]: Sending event: {'category': 'dbt', 'action': 'resource_counts', 'label': '23ee032b-8d56-4a19-b107-2fffc83e9042', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x10c74e220>]}
17:17:32.669514 [info ] [MainThread]: Found 177 models, 3786 tests, 0 snapshots, 0 analyses, 590 macros, 0 operations, 0 seed files, 98 sources, 0 exposures, 0 metrics
17:17:32.669952 [debug] [MainThread]: Sending event: {'category': 'dbt', 'action': 'runnable_timing', 'label': '23ee032b-8d56-4a19-b107-2fffc83e9042', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x10a585400>]}
17:17:32.827246 [info ] [MainThread]:
17:17:32.827842 [debug] [MainThread]: Acquiring new trino connection "master"
17:17:32.829450 [debug] [ThreadPool]: Acquiring new trino connection "list_ol_data_lake_qa"
17:17:32.844099 [debug] [ThreadPool]: Using trino connection "list_ol_data_lake_qa"
17:17:32.844393 [debug] [ThreadPool]: On list_ol_data_lake_qa: select distinct schema_name
    from ol_data_lake_qa.INFORMATION_SCHEMA.schemata
17:17:32.844678 [debug] [ThreadPool]: Opening a new connection, currently in state init
17:17:33.667961 [debug] [ThreadPool]: SQL status: SUCCESS in 0.82 seconds
17:17:33.670373 [debug] [ThreadPool]: On list_ol_data_lake_qa: Close
17:17:33.691363 [debug] [ThreadPool]: Acquiring new trino connection "list_ol_data_lake_qa_ol_warehouse_qa_staging"
17:17:33.701579 [debug] [ThreadPool]: Using trino connection "list_ol_data_lake_qa_ol_warehouse_qa_staging"
17:17:33.701928 [debug] [ThreadPool]: On list_ol_data_lake_qa_ol_warehouse_qa_staging: select
      table_catalog as database,
      table_name as name,
      table_schema as schema,
      case when table_type = 'BASE TABLE' then 'table'
           when table_type = 'VIEW' then 'view'
           else table_type
      end as table_type
    from ol_data_lake_qa.INFORMATION_SCHEMA.tables
    where table_schema = 'ol_warehouse_qa_staging'
17:17:33.702257 [debug] [ThreadPool]: Opening a new connection, currently in state closed
17:17:35.105744 [debug] [ThreadPool]: SQL status: SUCCESS in 1.4 seconds
17:17:35.110968 [debug] [ThreadPool]: On list_ol_data_lake_qa_ol_warehouse_qa_staging: ROLLBACK
17:17:35.111334 [debug] [ThreadPool]: On list_ol_data_lake_qa_ol_warehouse_qa_staging: Close
17:17:35.112522 [debug] [ThreadPool]: Acquiring new trino connection "list_ol_data_lake_qa_ol_warehouse_qa_intermediate"
17:17:35.119062 [debug] [ThreadPool]: Using trino connection "list_ol_data_lake_qa_ol_warehouse_qa_intermediate"
17:17:35.119438 [debug] [ThreadPool]: On list_ol_data_lake_qa_ol_warehouse_qa_intermediate: select
      table_catalog as database,
      table_name as name,
      table_schema as schema,
      case when table_type = 'BASE TABLE' then 'table'
           when table_type = 'VIEW' then 'view'
           else table_type
      end as table_type
    from ol_data_lake_qa.INFORMATION_SCHEMA.tables
    where table_schema = 'ol_warehouse_qa_intermediate'
17:17:35.119769 [debug] [ThreadPool]: Opening a new connection, currently in state closed
17:17:36.346300 [debug] [ThreadPool]: SQL status: SUCCESS in 1.23 seconds
17:17:36.350657 [debug] [ThreadPool]: On list_ol_data_lake_qa_ol_warehouse_qa_intermediate: ROLLBACK
17:17:36.351075 [debug] [ThreadPool]: On list_ol_data_lake_qa_ol_warehouse_qa_intermediate: Close
17:17:36.352335 [debug] [ThreadPool]: Acquiring new trino connection "list_ol_data_lake_qa_ol_warehouse_qa_mart"
17:17:36.361433 [debug] [ThreadPool]: Using trino connection "list_ol_data_lake_qa_ol_warehouse_qa_mart"
17:17:36.361782 [debug] [ThreadPool]: On list_ol_data_lake_qa_ol_warehouse_qa_mart: select
      table_catalog as database,
      table_name as name,
      table_schema as schema,
      case when table_type = 'BASE TABLE' then 'table'
           when table_type = 'VIEW' then 'view'
           else table_type
      end as table_type
    from ol_data_lake_qa.INFORMATION_SCHEMA.tables
    where table_schema = 'ol_warehouse_qa_mart'
17:17:36.362107 [debug] [ThreadPool]: Opening a new connection, currently in state closed
17:17:37.321953 [debug] [ThreadPool]: SQL status: SUCCESS in 0.96 seconds
17:17:37.328643 [debug] [ThreadPool]: On list_ol_data_lake_qa_ol_warehouse_qa_mart: ROLLBACK
17:17:37.328985 [debug] [ThreadPool]: On list_ol_data_lake_qa_ol_warehouse_qa_mart: Close
17:17:37.329969 [debug] [MainThread]: Sending event: {'category': 'dbt', 'action': 'runnable_timing', 'label': '23ee032b-8d56-4a19-b107-2fffc83e9042', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x10f277c10>]}
17:17:37.331228 [debug] [MainThread]: On master: COMMIT
17:17:37.331959 [info ] [MainThread]: Concurrency: 1 threads (target='qa')
17:17:37.332294 [info ] [MainThread]:
17:17:37.336559 [debug] [Thread-1  ]: Began running node model.open_learning.stg__edxorg__bigquery__mitx_courserun
17:17:37.337046 [info ] [Thread-1  ]: 1 of 1 START sql table model ol_warehouse_qa_staging.stg__edxorg__bigquery__mitx_courserun  [RUN]
17:17:37.338049 [debug] [Thread-1  ]: Acquiring new trino connection "model.open_learning.stg__edxorg__bigquery__mitx_courserun"
17:17:37.338397 [debug] [Thread-1  ]: Began compiling node model.open_learning.stg__edxorg__bigquery__mitx_courserun
17:17:37.338739 [debug] [Thread-1  ]: Compiling model.open_learning.stg__edxorg__bigquery__mitx_courserun
17:17:37.343570 [debug] [Thread-1  ]: Writing injected SQL for node "model.open_learning.stg__edxorg__bigquery__mitx_courserun"
17:17:37.344271 [debug] [Thread-1  ]: finished collecting timing info
17:17:37.344630 [debug] [Thread-1  ]: Began executing node model.open_learning.stg__edxorg__bigquery__mitx_courserun
17:17:37.375304 [debug] [Thread-1  ]: Using trino connection "model.open_learning.stg__edxorg__bigquery__mitx_courserun"
17:17:37.375614 [debug] [Thread-1  ]: On model.open_learning.stg__edxorg__bigquery__mitx_courserun: drop table if exists ol_data_lake_qa.ol_warehouse_qa_staging.stg__edxorg__bigquery__mitx_courserun__dbt_tmp
17:17:37.376010 [debug] [Thread-1  ]: Opening a new connection, currently in state closed
17:17:38.096751 [debug] [Thread-1  ]: SQL status: SUCCESS in 0.72 seconds
17:17:38.102278 [debug] [Thread-1  ]: Using trino connection "model.open_learning.stg__edxorg__bigquery__mitx_courserun"
17:17:38.102617 [debug] [Thread-1  ]: On model.open_learning.stg__edxorg__bigquery__mitx_courserun: drop table if exists ol_data_lake_qa.ol_warehouse_qa_staging.stg__edxorg__bigquery__mitx_courserun__dbt_backup
17:17:38.798883 [debug] [Thread-1  ]: SQL status: SUCCESS in 0.7 seconds
17:17:38.818403 [debug] [Thread-1  ]: Writing runtime sql for node "model.open_learning.stg__edxorg__bigquery__mitx_courserun"
17:17:38.822168 [debug] [Thread-1  ]: Using trino connection "model.open_learning.stg__edxorg__bigquery__mitx_courserun"
17:17:38.822530 [debug] [Thread-1  ]: On model.open_learning.stg__edxorg__bigquery__mitx_courserun: create table ol_data_lake_qa.ol_warehouse_qa_staging.stg__edxorg__bigquery__mitx_courserun__dbt_tmp

  as (
    -- MITx Courses Information sourced from edx.org

-- Note that course id here is always in old format as {org}/{course number}/{run} as IR converts the new format
-- to old format in order to preserve functionality of old apps
-- e.g course_id: "MITx/14.73x/3T2020" is the same course run "course-v1:MITx+14.73x+3T2020" in MITxOnline

with source as (

    select * from ol_data_lake_qa.ol_warehouse_qa_raw.raw__irx__edxorg__bigquery__mitx_course

)

, renamed as (

    select
        course_number
        , course_title as courserun_title
        , semester as courserun_semester
        , url as courserun_url
        , institution as courserun_institution
        , instructors as courserun_instructors
        , registration_open as courserun_enrollment_start_date
        , course_launch as courserun_start_date
        , course_wrap as courserun_end_date
        , replace(course_id, 'ESD.SCM1x', 'CTL.SC1x') as courserun_readable_id
        , coalesce(self_paced = 1, false) as courserun_is_self_paced
    from source

)

select * from renamed
  )
17:17:40.430391 [debug] [Thread-1  ]: Trino adapter: Trino query id: 20230209_171738_81550_imgyu
17:17:40.430912 [debug] [Thread-1  ]: Trino adapter: Trino error: TrinoExternalError(type=EXTERNAL, name=HIVE_CURSOR_ERROR, message="io.trino.hive.$internal.org.codehaus.jackson.JsonParseException: Current token (VALUE_STRING) not numeric, can not use numeric value accessors
 at [Source: [B@4b1b5ecc; line: 1, column: 308]", query_id=20230209_171738_81550_imgyu)
17:17:40.431647 [debug] [Thread-1  ]: finished collecting timing info
17:17:40.432085 [debug] [Thread-1  ]: On model.open_learning.stg__edxorg__bigquery__mitx_courserun: ROLLBACK
17:17:40.432470 [debug] [Thread-1  ]: On model.open_learning.stg__edxorg__bigquery__mitx_courserun: Close
17:17:40.433598 [debug] [Thread-1  ]: Database Error in model stg__edxorg__bigquery__mitx_courserun (models/staging/edxorg/stg__edxorg__bigquery__mitx_courserun.sql)
  TrinoExternalError(type=EXTERNAL, name=HIVE_CURSOR_ERROR, message="io.trino.hive.$internal.org.codehaus.jackson.JsonParseException: Current token (VALUE_STRING) not numeric, can not use numeric value accessors
   at [Source: [B@4b1b5ecc; line: 1, column: 308]", query_id=20230209_171738_81550_imgyu)
  compiled Code at target/run/open_learning/models/staging/edxorg/stg__edxorg__bigquery__mitx_courserun.sql
Traceback (most recent call last):
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/dbt/adapters/trino/connections.py", line 368, in exception_handler
    yield
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/dbt/adapters/sql/connections.py", line 70, in add_query
    cursor.execute(sql, bindings)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/dbt/adapters/trino/connections.py", line 330, in execute
    result = self._cursor.execute(sql, params=bindings)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/trino/dbapi.py", line 435, in execute
    result = self._query.execute()
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/trino/client.py", line 758, in execute
    self._result.rows += self.fetch()
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/trino/client.py", line 773, in fetch
    status = self._request.process(response)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/trino/client.py", line 574, in process
    raise self._process_error(response["error"], response.get("id"))
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/trino/client.py", line 543, in _process_error
    raise exceptions.TrinoExternalError(error, query_id)
trino.exceptions.TrinoExternalError: TrinoExternalError(type=EXTERNAL, name=HIVE_CURSOR_ERROR, message="io.trino.hive.$internal.org.codehaus.jackson.JsonParseException: Current token (VALUE_STRING) not numeric, can not use numeric value accessors
 at [Source: [B@4b1b5ecc; line: 1, column: 308]", query_id=20230209_171738_81550_imgyu)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/dbt/task/base.py", line 385, in safe_run
    result = self.compile_and_execute(manifest, ctx)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/dbt/task/base.py", line 338, in compile_and_execute
    result = self.run(ctx.node, manifest)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/dbt/task/base.py", line 429, in run
    return self.execute(compiled_node, manifest)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/dbt/task/run.py", line 281, in execute
    result = MacroGenerator(
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/dbt/clients/jinja.py", line 326, in __call__
    return self.call_macro(*args, **kwargs)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/dbt/clients/jinja.py", line 253, in call_macro
    return macro(*args, **kwargs)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/jinja2/runtime.py", line 763, in __call__
    return self._invoke(arguments, autoescape)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/jinja2/runtime.py", line 777, in _invoke
    rv = self._func(*arguments)
  File "<template>", line 96, in macro
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/jinja2/sandbox.py", line 393, in call
    return __context.call(__obj, *args, **kwargs)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/jinja2/runtime.py", line 298, in call
    return __obj(*args, **kwargs)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/dbt/clients/jinja.py", line 326, in __call__
    return self.call_macro(*args, **kwargs)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/dbt/clients/jinja.py", line 253, in call_macro
    return macro(*args, **kwargs)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/jinja2/runtime.py", line 763, in __call__
    return self._invoke(arguments, autoescape)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/jinja2/runtime.py", line 777, in _invoke
    rv = self._func(*arguments)
  File "<template>", line 52, in macro
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/jinja2/sandbox.py", line 393, in call
    return __context.call(__obj, *args, **kwargs)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/jinja2/runtime.py", line 298, in call
    return __obj(*args, **kwargs)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/dbt/adapters/base/impl.py", line 278, in execute
    return self.connections.execute(sql=sql, auto_begin=auto_begin, fetch=fetch)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/dbt/adapters/trino/connections.py", line 481, in execute
    _, cursor = self.add_query(sql, auto_begin)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/dbt/adapters/trino/connections.py", line 460, in add_query
    connection, cursor = parent.add_query(
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/dbt/adapters/sql/connections.py", line 78, in add_query
    return connection, cursor
  File "/Users/sar/.pyenv/versions/3.9.12/lib/python3.9/contextlib.py", line 137, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/Users/sar/Library/Caches/pypoetry/virtualenvs/ol-data-platform-IaJixTul-py3.9/lib/python3.9/site-packages/dbt/adapters/trino/connections.py", line 379, in exception_handler
    raise DatabaseException(msg)
dbt.exceptions.DatabaseException: Database Error in model stg__edxorg__bigquery__mitx_courserun (models/staging/edxorg/stg__edxorg__bigquery__mitx_courserun.sql)
  TrinoExternalError(type=EXTERNAL, name=HIVE_CURSOR_ERROR, message="io.trino.hive.$internal.org.codehaus.jackson.JsonParseException: Current token (VALUE_STRING) not numeric, can not use numeric value accessors
   at [Source: [B@4b1b5ecc; line: 1, column: 308]", query_id=20230209_171738_81550_imgyu)
  compiled Code at target/run/open_learning/models/staging/edxorg/stg__edxorg__bigquery__mitx_courserun.sql
17:17:40.442074 [debug] [Thread-1  ]: Sending event: {'category': 'dbt', 'action': 'run_model', 'label': '23ee032b-8d56-4a19-b107-2fffc83e9042', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x10f3c9e50>]}
17:17:40.442718 [error] [Thread-1  ]: 1 of 1 ERROR creating sql table model ol_warehouse_qa_staging.stg__edxorg__bigquery__mitx_courserun  [ERROR in 3.10s]
17:17:40.443275 [debug] [Thread-1  ]: Finished running node model.open_learning.stg__edxorg__bigquery__mitx_courserun
17:17:40.445164 [debug] [MainThread]: Acquiring new trino connection "master"
17:17:40.446138 [debug] [MainThread]: On master: COMMIT
17:17:40.446891 [info ] [MainThread]:
17:17:40.447246 [info ] [MainThread]: Finished running 1 table model in 0 hours 0 minutes and 7.62 seconds (7.62s).
17:17:40.447593 [debug] [MainThread]: Connection 'master' was properly closed.
17:17:40.447920 [debug] [MainThread]: Connection 'model.open_learning.stg__edxorg__bigquery__mitx_courserun' was properly closed.
17:17:40.850601 [info ] [MainThread]:
17:17:40.850978 [info ] [MainThread]: Completed with 1 error and 0 warnings:
17:17:40.851318 [info ] [MainThread]:
17:17:40.851653 [error] [MainThread]: Database Error in model stg__edxorg__bigquery__mitx_courserun (models/staging/edxorg/stg__edxorg__bigquery__mitx_courserun.sql)
17:17:40.851970 [error] [MainThread]:   TrinoExternalError(type=EXTERNAL, name=HIVE_CURSOR_ERROR, message="io.trino.hive.$internal.org.codehaus.jackson.JsonParseException: Current token (VALUE_STRING) not numeric, can not use numeric value accessors
17:17:40.852273 [error] [MainThread]:    at [Source: [B@4b1b5ecc; line: 1, column: 308]", query_id=20230209_171738_81550_imgyu)
17:17:40.852572 [error] [MainThread]:   compiled Code at target/run/open_learning/models/staging/edxorg/stg__edxorg__bigquery__mitx_courserun.sql
17:17:40.852884 [info ] [MainThread]:
17:17:40.853230 [info ] [MainThread]: Done. PASS=0 WARN=0 ERROR=1 SKIP=0 TOTAL=1
17:17:40.853660 [debug] [MainThread]: Sending event: {'category': 'dbt', 'action': 'invocation', 'label': 'end', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x10f3c7d30>, <snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x10f3a5d90>, <snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x10f764b50>]}
17:17:40.854105 [debug] [MainThread]: Flushing usage events
```
### Related Issues

(Optional)

### Screenshot or Screencast

(Optional)

**Comment by shaidar** [comment]
Chatted with Starburst folks about it but they weren't sure what could be causing this. I did come across this issue https://github.com/trinodb/trino/issues/12866 that seems similar to what we're running into. Tried querying QA in Athena, and that works, however it fails in the Starburst UI which in my mind is pointing back to data syncing and the glue connector and how it's formatting the data that it uploads to S3.
